### ISSUE TITLE
Biospheres required clientside

### ISSUE THREAD

**Issue by TheLoneWolfling** [BrainSlugs83/NewBiospheresMod#19]
It would be wonderful if Biospheres was only required serverside.

Unfortunately, Forge 1.7.10 doesn't like vanilla clients, so it's unlikely to happen. But still, figured I may as well toss the idea out there.

**Comment by BrainSlugs83** [comment]
Someone mentioned the idea of making a bukkit plugin too -- would that solve this issue?

I need to do some research here (I read that bukkit was dead, but also that it wasn't, but then, also that it was...) -- but if there's a way to do it (forge or not) -- in a way that would be useful for server runners, I think we should.  (I eventually want to run this mod on my own server.)

I'm thinking that most of the guts of the mod can be abstracted out, and that there can just be a couple of wrapper/framework/interface classes that just kind of hook everything in for a given framework.

That way we could target multiple frameworks (or even versions of frameworks), such as forge/bukkit/joe's-fantastic-fictional-framework-that-I-just-made-up/etc.

This is kind of a long term goal I think (probably will approach it with the upgrade to MC 1.8, since there will be some required refactoring with the upgrade anyway).  But in the meanwhile, if you, (or anyone else), have advice or guidance (or links to such), please don't hesitate to post them here.

**Comment by TheLoneWolfling** [comment]
For Forge itself on 1.7.10, this blocks on [this](https://github.com/MinecraftForge/FML/issues/510) FML bug. Namely that vanilla clients trying to connect to a Forge server just hang.

However, supposedly Forge 1.8 will (does?) (supposedly) support vanilla clients [link](http://www.reddit.com/r/feedthebeast/comments/2fn3dk/forge_18_to_support_vanilla_clients_and_will/). Don't know about the current state of it though.

I don't know enough to have much of any other advice, beyond "break it out into generating 16x16x16 chunks" - that way it'll be easy to add Cubic Chunk compatibility as well, if/when that ever happens.

**Comment by BrainSlugs83** [comment]
Yeah, I'm just thinking -- if I refactor the code to use more of a modified "provider" pattern, I could make it work with other frameworks as well (i.e. make a version that works with Bukkit *instead* of forge -- but I don't know if bukkit is the right framework to target).

Essentially have one folder (or "package") for all the Forge hooks, and another for Bukkit (or whatever other framework, etc.) -- not sure how feasible this is or how much glue it will need.

As for cubic chunks (or other chunking strategies), the models and logic of the sphere generation is already decently divorced from the chunk size, so changes like that shouldn't be too hard to deal with I think. -- that's kind of a bridge that we can burn when we get to it. :-)

I guess I just [gotta let my mind fester for on this for a bit](http://www.subzin.com/quotes/S1634856098/Adventure+Time+with+Finn+%26+Jake/Puhoy/Gotta+let+my+mind+fester+a+bit,+you+know%3F).
